android/app/src/main/java/com/zbaymobile/Utils/Utils.kt

```diff
@@ -0,0 +1,104 @@
+package com.zbaymobile.Utils
+
+import android.content.Context
+import android.os.Build
+import android.util.Log
+import com.zbaymobile.Utils.Const.TAG_NOTICE
+import java.io.BufferedReader
+import java.io.File
+import java.io.InputStreamReader
+import java.net.ConnectException
+import java.net.InetSocketAddress
+import java.net.Socket
+
+
+object Utils {
+
+    fun getNativeLibraryDir(context: Context): String? {
+        val appInfo = context.applicationInfo
+        return appInfo.nativeLibraryDir
+    }
+
+    fun getArch(): String {
+        var folder = Build.CPU_ABI
+        val javaArch = System.getProperty("os.arch")
+        if (javaArch != null && javaArch.contains("686")) {
+            folder = "x86"
+        }
+        // return folder
+        return "arm64-v8a"
+    }
+
+    fun exec(dir: File, command: Array<String>, env: Map<String, String>?): Process {
+        val pb = ProcessBuilder(*command)
+        val _env: MutableMap<String, String> = pb.environment()
+        env?.map {
+            _env[it.key] = it.value
+        }
+        return pb
+            .directory(dir)
+            .start()
+    }
+
+    fun getOutput(process: Process): Int {
+        val stdOut = BufferedReader(InputStreamReader(process.inputStream))
+        val stdError = BufferedReader(InputStreamReader(process.errorStream))
+
+        object: Thread() {
```

Comment:
few files up I've seen you did:
```
   process.stderr.map {            Log.d("TOR_ERR", it)        }       
   process.stdout.map {            Log.d(TAG_TOR, it)        }
```

which looks nicer than these threads here

Reply:
Ok, cool

Reply:
Also keep in mind that unlike Tor, Waggle's process never finishes so there is no possibility to just get access to stdout and stderr without conditionally looping them in separate threads